src/main/scala/io/iohk/ethereum/domain/Blockchain.scala

```diff
@@ -280,6 +289,7 @@ class BlockchainImpl(
     receiptStorage.remove(blockHash)
     maybeTxList.foreach(removeTxsLocations)
     maybeBlockHeader.foreach{ h =>
+      rollbackState(h.number)
```

Comment:
No, because it rollback state changes, ie, undo whatever has been done. Maybe `rollbackState` is not a good name and `rollbackStateChanges` is more appropriate, what do you think?

Reply:
So if I understood correctly it rollbacks state changes made by block we are currently deleting, and not rollback state changes to some block? (if yes then it could be good to rename function to sth like 
`rollbackStateChangesMadeByBlock` or something similar)